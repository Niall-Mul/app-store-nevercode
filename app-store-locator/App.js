/**
 * Copyright 2018-present Singlepoint. All Rights Reserved.
 *
 * @flow
 */

import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { ConnectedRouter, routerMiddleware } from 'react-router-redux';
import ReduxThunk from 'redux-thunk';
import { apiMiddleware } from 'redux-api-middleware';
import { createLogger } from 'redux-logger';
import { Iterable } from 'immutable';
import { I18nextProvider } from 'react-i18next';
import { translate } from 'react-i18next';
import { createStackNavigator } from 'react-navigation';
import transitionConfig from './transitionConfig';
import reducers from './reducers/RootReducer';
import { i18n } from '@vodafone/core-app/i18n';
import { containers } from '@vodafone/core-app/styles';
import routes from './routes.js';

const stateTransformer = state => {
  if (Iterable.isIterable(state)) return state.toJS();
  else return state;
};
const reduxLogger = createLogger({
  collapsed: true,
  stateTransformer,
});

const isDebuggingEnabled = typeof atob !== 'undefined';
const middleware = [
  apiMiddleware,
  ReduxThunk,
  isDebuggingEnabled && reduxLogger,
].filter(Boolean);
const store = createStore(reducers, applyMiddleware(...middleware));

const RootStack = createStackNavigator(routes, {
  initialRouteName: 'Home',
  headerMode: 'none',
  transitionConfig,
});

interface Props {}

class App extends React.Component<Props> {
  render() {
    return (
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <RootStack />
        </I18nextProvider>
      </Provider>
    );
  }
}

export default App;
