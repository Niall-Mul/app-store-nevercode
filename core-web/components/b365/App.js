/**
 * Copyright 2015-present Singlepoint. All Rights Reserved.
 *
 * @flow
 */
import React from 'react';
import { Route, withRouter } from 'react-router-dom';
import { Container, Header, Image } from 'semantic-ui-react';

import HomePage from './HomePage';
import SecureLoginPage from './SecureLoginPage';
import YourAccountPage from './YourAccountsPage';


interface Props {
  assetPath: string;
  basePath: string;
  logo: string;
  history: any;
}

class AppB365 extends React.Component<Props> {
  render() {
    let { basePath, logo } = this.props;

    return (
      <div className="appContainer">
        <Header attached="top" as="h1" className={'main b365-header'}>
          <Container textAlign="center">
            <Image src={logo} className="headerLogo" />
          </Container>
        </Header>

        <main>
          <Route exact path={basePath} render={() => <HomePage {...this.props} />} />
          <Route exact path={`${basePath}/secure-login`} render={() => <SecureLoginPage {...this.props} />} />
          <Route exact path={`${basePath}/your-accounts`} render={() => <YourAccountPage {...this.props} />} />
        </main>
      </div>
    );
  }
}


export default withRouter(AppB365);
