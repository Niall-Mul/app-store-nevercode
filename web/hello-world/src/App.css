.App {
  text-align: center;
}

.App-logos {
  height: 5vmin;
  vertical-align: middle;
  margin-right: 2px;
  float: right;
}

.App-logo {
  height: 15vmin;
  vertical-align: middle;
  margin-right: 20px;
}

.App-header-bar {
  vertical-align: middle;
}

.logoBar {
  vertical-align: middle;
}

.App-react-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 15vmin;
  vertical-align: bottom;
}

.App-redux-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 10vmin;
  vertical-align: top;
  margin-top: 15px;
}

.App-nodejs-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 10vmin;
  vertical-align: top;
  margin-top: 20px;
  margin-left: 35px;
}

.App-content {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  padding-top: 20px;
}

.App-link {
  color: #61dafb;
}

.App-margin-Large {
  margin-top: 15px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
