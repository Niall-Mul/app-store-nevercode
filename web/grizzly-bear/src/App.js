/**
 * Copyright 2015-present Singlepoint. All Rights Reserved.
 *
 * @flow
 */

import React from 'react';
import { Redirect } from 'react-router';
import { Route, withRouter } from 'react-router-dom';

import CarInsuranceSubmitClaimPage from './pages/insurances/car/CarInsuranceSubmitClaimPage';

import {
  Dashboard,
  MyInsuranceProfilePage,
  PlaceholderSFC,
  LoginPage,
  Logout,
  InsurancePage,
  InsuranceAmend,
  InsuranceRenewal,
  InsuranceRenew,
  InsuranceClaims,
  InsuranceMakeAClaim,
  CarInsuranceQuote,
  CarInsuranceAddDriver,
  CarInsuranceDriverDetails,
  InsuranceManage,
  HomeInsuranceQuote,
  TravelInsuranceQuote,
  QuoteDetailsPage,
  MenuNavLink,
  QuotePayment,
  PaymentConfirmation,
  PaymentDocumentsPage,
  QuoteDetailsExpandedPage,
} from '@vodafone/core-web/boi';

import {
  Header,
  Image,
  Container,
  Menu,
  Icon,
  Sidebar,
  Segment,
  Responsive,
  Button,
} from 'semantic-ui-react';

import './App.css';

interface Props {
  logo?: string;
  className?: string;
  basePath: string;
  history: any;
}

interface State {
  visible: boolean;
}

const menuItems = [
  {
    link: '/home',
    text: 'Home',
    icon: 'home',
  },
  {
    link: '#',
    text: 'My Account',
  },
  {
    link: '#',
    text: 'Manage my Policy',
  },
  {
    link: '/my-insurance',
    text: 'My insurance profile',
  },
  {
    link: '/logout',
    text: 'Logout',
  },
  {
    link: '#',
    text: 'Help',
  },
];

class App extends React.Component<Props, State> {
  state = {
    visible: false,
  };

  handleToggle = () => {
    this.setState({ visible: !this.state.visible });
  };

  handleSidebarHide = () => {
    this.setState({ visible: false });
  };

  handleGoBack = () => {
    this.props.history.goBack();
  };

  render() {
    const menuOptions = menuItems.map((item, index) => (
      <Menu.Item
        key={index}
        as={MenuNavLink}
        to={item.link != '#' ? (this.props.basePath + item.link) : '#'}
        onClick={this.handleSidebarHide}
        className="menuItem"
      >
        {item.icon ? (
          <Responsive minWidth={Responsive.onlyTablet.minWidth}>
            <Icon name={item.icon} color="grey" />
          </Responsive>
        ) : null}
        {item.text}
      </Menu.Item>
    ));

    const isHomePage = /^\/wallet\/home\/?$/.test(this.props.history.location.pathname);

    return (
      <div
        className={
          'appContainer' +
          (this.props.className ? ' ' + this.props.className : '')
        }
      >
        <Sidebar.Pushable
          as={Segment}
          style={this.state.visible ? { overflowY: 'hidden' } : {}}
        >
          <Sidebar
            as={Menu}
            animation="overlay"
            icon="labeled"
            onHide={this.handleSidebarHide}
            vertical
            visible={this.state.visible}
            width="wide"
            direction="right"
          >
            <Menu.Item className="menuItem menuTitle">Menu</Menu.Item>
            {menuOptions}
            {/* <Menu.Item className="menuItem menuImage">
              <Image src={'/assets/bank_of_ireland-logo.png'} />
            </Menu.Item> */}
          </Sidebar>

          <Sidebar.Pusher dimmed={this.state.visible}>
            {this.props.history.location.pathname !== '/wallet' &&
              this.props.history.location.pathname !== '/wallet/' ? (
                <Header
                  attached="top"
                  as="h1"
                  className={
                    this.props.history.location.pathname !== '/wallet/'
                      ? 'main'
                      : 'main small'
                  }
                >
                  <Container>
                    {isHomePage ? null : (
                      <Responsive
                        as={Segment}
                        {...Responsive.onlyMobile}
                        className="headerBackButton"
                      >
                        <a href="#" onClick={this.handleGoBack}>
                          <Icon name="arrow left" size="large"></Icon>
                        </a>
                      </Responsive>
                    )}
                    <Image src={this.props.logo} className="headerLogo" />
                    <span className="text">Insurance Wallet</span>
                    <Responsive
                      as={Segment}
                      {...Responsive.onlyMobile}
                      className="collapseMenu"
                    >
                      <Button
                        onClick={this.handleToggle}
                        className="collapseMenuButton"
                      >
                        <Icon name="bars" className="collapseMenuIcon" />
                      </Button>
                    </Responsive>
                  </Container>
                  {this.props.history.location.pathname !== '/wallet/' ? (
                    <Responsive
                      as={Segment}
                      minWidth={Responsive.onlyTablet.minWidth}
                      style={{
                        padding: 0,
                        margin: 0,
                        border: 0,
                      }}
                    >
                      <Menu borderless className="mainMenu">
                        <Container>{menuOptions}</Container>
                      </Menu>
                    </Responsive>
                  ) : null}
                </Header>
              ) : null}

            <main>
              <Route
                exact
                path="/"
                render={() => <Redirect to={this.props.basePath} />}
              />
              <Route
                exact
                path={this.props.basePath}
                render={() => (
                  <LoginPage
                    className="SP-LoginPage"
                    basePath={this.props.basePath}
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/logout'}
                render={() => <Logout />}
              />
              <Route
                exact
                path={this.props.basePath + '/home'}
                render={() => (
                  <Dashboard
                    className="SP-PageContent"
                    basePath={this.props.basePath}
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/my-insurance'}
                render={() => (
                  <MyInsuranceProfilePage
                    label="My insurance profile"
                    className="SP-PageContent"
                    basePath={this.props.basePath}
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/help'}
                render={() => (
                  <PlaceholderSFC
                    label="Contact Us"
                    className="SP-PageContent"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/settings'}
                render={() => (
                  <PlaceholderSFC label="Settings" className="SP-PageContent" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance'}
                render={() => (
                  <InsurancePage className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/renewal'}
                render={() => (
                  <InsuranceRenewal className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/amend'}
                render={() => (
                  <InsuranceAmend className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/manage'}
                render={() => (
                  <InsuranceManage className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/manage/renew'}
                render={() => (
                  <InsuranceRenew className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/amend/driver-details'}
                render={() => (
                  <CarInsuranceDriverDetails className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/amend/driver-details/add-driver'}
                render={() => (
                  <CarInsuranceAddDriver className="SP-InsurancePage" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/quotes'}
                render={() => (
                  <CarInsuranceQuote className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/manage/claims'}
                render={() => (
                  <InsuranceClaims className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/make-a-claim'}
                render={() => (
                  <InsuranceMakeAClaim className="SP-InsurancePage" quoteType="car" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/submit-claim'}
                render={() => <CarInsuranceSubmitClaimPage />}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/quote-results'}
                render={() => (
                  <QuoteDetailsPage
                    label="Quote Results"
                    className="SP-PageContent"
                    quoteType="car"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/expanded'}
                render={() => (
                  <QuoteDetailsExpandedPage
                    className="SP-PageContent"
                    quoteType="car"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/renewal/quote-results'}
                render={() => (
                  <QuoteDetailsPage
                    label="Quote Results"
                    className="SP-PageContent"
                    quoteType="carRenewal"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/renewal/expanded'}
                render={() => (
                  <QuoteDetailsExpandedPage
                    className="SP-PageContent"
                    quoteType="carRenewal"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/renewal/payment'}
                render={() => <QuotePayment className="SP-PageContent" />}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/renewal/confirm'}
                render={() => (
                  <PaymentConfirmation className="SP-PageContent" quoteType="carRenewal" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/payment'}
                render={() => <QuotePayment className="SP-PageContent" />}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/confirm'}
                render={() => (
                  <PaymentConfirmation className="SP-PageContent" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/car-insurance/documents'}
                render={() => (
                  <PaymentDocumentsPage
                    className="SP-PageContent"
                    quoteType="car"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance'}
                render={() => (
                  <InsurancePage
                    className="SP-InsurancePage"
                    quoteType="travel"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/quotes'}
                render={() => (
                  <TravelInsuranceQuote className="SP-InsurancePage" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/manage'}
                render={() => (
                  <InsuranceManage className="SP-InsurancePage" quoteType="travel" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/quote-results'}
                render={() => (
                  <QuoteDetailsPage
                    label="Quote Results"
                    className="SP-PageContent"
                    quoteType="travel"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/expanded'}
                render={() => (
                  <QuoteDetailsExpandedPage
                    className="SP-PageContent"
                    quoteType="travel"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/payment'}
                render={() => <QuotePayment className="SP-PageContent" />}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/confirm'}
                render={() => (
                  <PaymentConfirmation className="SP-PageContent" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/travel-insurance/documents'}
                render={() => (
                  <PaymentDocumentsPage
                    className="SP-PageContent"
                    quoteType="travel"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/pet-insurance'}
                render={() => (
                  <PlaceholderSFC
                    label="Pet Insurance"
                    className="SP-PageContent"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance'}
                render={() => (
                  <InsurancePage
                    className="SP-InsurancePage"
                    quoteType="home"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/quotes'}
                render={() => (
                  <HomeInsuranceQuote className="SP-InsurancePage" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/quote-results'}
                render={() => (
                  <QuoteDetailsPage
                    label="Quote Results"
                    className="SP-PageContent"
                    quoteType="home"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/expanded'}
                render={() => (
                  <QuoteDetailsExpandedPage
                    className="SP-PageContent"
                    quoteType="home"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/payment'}
                render={() => <QuotePayment className="SP-PageContent" />}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/confirm'}
                render={() => (
                  <PaymentConfirmation className="SP-PageContent" />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/home-insurance/documents'}
                render={() => (
                  <PaymentDocumentsPage
                    className="SP-PageContent"
                    quoteType="home"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/health-insurance'}
                render={() => (
                  <PlaceholderSFC
                    label="Health Insurance"
                    className="SP-PageContent"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/life-insurance'}
                render={() => (
                  <PlaceholderSFC
                    label="Life Insurance"
                    className="SP-PageContent"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/van-insurance'}
                render={() => (
                  <PlaceholderSFC
                    label="Van Insurance"
                    className="SP-PageContent"
                  />
                )}
              />
              <Route
                exact
                path={this.props.basePath + '/business-insurance'}
                render={() => (
                  <PlaceholderSFC
                    label="Business Insurance"
                    className="SP-PageContent"
                  />
                )}
              />
            </main>
          </Sidebar.Pusher>
        </Sidebar.Pushable>
      </div>
    );
  }
}

export default withRouter(App);
