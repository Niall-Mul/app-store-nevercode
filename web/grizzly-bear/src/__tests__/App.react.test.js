import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import App from '../App.react';

describe('App Component', () => {
  let root;
  let props;
  let mountedBurgerMenu;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    const output = renderer.create(<App />).toJSON();
    expect(output).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});
