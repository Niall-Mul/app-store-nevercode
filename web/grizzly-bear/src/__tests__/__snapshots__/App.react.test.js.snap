// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div>
  <h1
    className="ui top attached header App-header-bar"
  >
    <div>
      <img
        alt="Vodafone"
        className="App-logo"
        src="/themes/vodafone/assets/Vodafone.png"
      />
      Hello World
      <div
        style={
          Object {
            "float": "right",
          }
        }
      >
        <img
          alt="React"
          className="App-react-logo"
          src="logo.svg"
        />
        <img
          alt="Redux"
          className="App-redux-logo"
          src="redux.svg"
        />
        <img
          alt="Redux"
          className="App-nodejs-logo"
          src="/themes/vodafone/assets/nodejs.png"
        />
      </div>
    </div>
  </h1>
  <div
    className="ui left aligned container App-content"
  >
    <div
      className="ui cards"
    >
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Counter
          </div>
          <div
            className="meta"
          >
            Redux Counter
          </div>
          <div
            className="description"
          >
            <div
              style={
                Object {
                  "paddingTop": "20px",
                }
              }
            >
              <h2
                className="ui header"
              >
                <i
                  aria-hidden="true"
                  className="stopwatch icon"
                />
                 
                0
              </h2>
              <button
                className="ui primary button"
                disabled={undefined}
                onClick={[Function]}
                role="button"
                tabIndex={undefined}
              >
                Increase
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            API
          </div>
          <div
            className="meta"
          >
            Spring Boot API
          </div>
          <div
            className="description"
          >
            <div>
              <button
                className="ui primary button"
                disabled={undefined}
                onClick={[Function]}
                role="button"
                tabIndex={undefined}
              >
                Invoke SpringBoot API
              </button>
              <div
                className="App-margin-Large"
              />
              <div
                className="App-margin-Large"
              >
                <div
                  className="ui centered inline loader"
                />
              </div>
              <div />
              <div
                className="ui feed"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
